_source/_posts/2023-07-27-spring-graphql-react.md

```diff
@@ -0,0 +1,1794 @@
+c.incorporationDate = Date(Datetime({epochSeconds: apoc.date.parse(row.IncorporationDate,'s','dd/MM/yyyy')})),
+c.SIC = row.`SICCode.SicText_1`,
+c.countryOfOrigin = row.CountryOfOrigin,
+c.status = row.CompanyStatus,
+c.category = row.CompanyCategory;
+```
+
+__V006__Land.cypher__
+```cypher
+LOAD CSV WITH HEADERS FROM "file:///LandOwnershipAmericans.csv" AS row
+MATCH (c:Company {companyNumber: row.`Company Registration No. (1)`})
+MERGE (p:Property {titleNumber: row.`Title Number`})
+SET p.address = row.`Property Address`,
+p.county  = row.County,
+p.price   = toInteger(row.`Price Paid`),
+p.district = row.District
+MERGE (c)-[r:OWNS]->(p)
+WITH row, c,r,p WHERE row.`Date Proprietor Added` IS NOT NULL
+SET r.date = Date(Datetime({epochSeconds: apoc.date.parse(row.`Date Proprietor Added`,'s','dd-MM-yyyy')}));
+CREATE INDEX FOR (c:Company) ON c.incorporationDate;
+```
+{% endraw %}
+
+Update `application.properties` and add the following properties:
+
+__application.properties__
+```properties
+...
+spring.graphql.graphiql.enabled=true
+spring.graphql.schema.introspection.enabled=true
+org.neo4j.migrations.transaction-mode=PER_STATEMENT
+spring.neo4j.uri=bolt://localhost:7687
+spring.neo4j.authentication.username=neo4j
+
+spring.graphql.cors.allowed-origins=http://localhost:3000
+```
+
+Create a `.env` file in the project root to store the Neo4j credentials:
+
+__.env__
+```shell
+export SPRING_NEO4J_AUTHENTICATION_PASSWORD=verysecret
+```
+
+Download the following seed files to some folder:
+
+- [CompanyDataAmericans](https://guides.neo4j.com/ukcompanies/data/CompanyDataAmericans.csv)
+- [LandOwnershipAmericans](https://guides.neo4j.com/ukcompanies/data/LandOwnershipAmericans.csv)
+- [PSCAmericans.csv](https://guides.neo4j.com/ukcompanies/data/PSCAmericans.csv)
+
+Create the folder `src/main/docker` and create a file `neo4j.yml` there, with the following content:
+
+__neo4j.yml__
+```yml
+# This configuration is intended for development purpose, it's **your** responsibility to harden it for production
+name: companies
+services:
+  neo4j:
+    image: neo4j:5
+    volumes:
+      - <csv-folder>:/var/lib/neo4j/import
+    environment:
+      - NEO4J_AUTH=neo4j/${NEO4J_PASSWORD}
+      - NEO4JLABS_PLUGINS=["apoc"]
+    # If you want to expose these ports outside your dev PC,
+    # remove the "127.0.0.1:" prefix
+    ports:
+      - '127.0.0.1:7474:7474'
+      - '127.0.0.1:7687:7687'
+    healthcheck:
+      test: ['CMD', 'wget', 'http://localhost:7474/', '-O', '-']
+      interval: 5s
+      timeout: 5s
+      retries: 10
+```
+
+Create the file `src/main/docker/.env` with the following content:
+
+__.env__
+```shell
+NEO4J_PASSWORD=verysecret
+```
+
+As you can see the compose file will mount `<csv-folder>` to a `/var/lib/neo4j/import` volume, making the content accessible from the running neo4j container.
+Replace `<csv-folder>` with the path to the CSV files downloaded before.
```

Comment:
For me, the `<csv-folder>` is `~/Downloads`, so that's what I put in `neo4j.yml`. I believe this is the default on macOS and Linux.

Reply:
Added a comment to use an empty folder.